### ISSUE TITLE
No comments added to issue on from footer commit or missing action from github parser

### ISSUE THREAD

**Issue by juancarlosjr97** [release-it/release-it#1179]
## Issue

Identifying the issue by release-it to write the comment 

## Diagnostic (so far)

So, if I have a group of commits using the `@release-it/conventional-changelog` with the preset `conventionalcommits ` (for example here https://github.com/juancarlosjr97/relese-automation-test/blob/main/.release-it.json#L28) such as the following commits

```
commit 59a134ec410d84abc8e87dba2d33dd23b4531003
feat: changes
    
#1
```

```
commit 74415e7ef1c3c6c98768b606a98fbdcb09830562
feat: changes
#1
```

or 

```
commit b5d89f49976000275fb1e7936f37d5dd7487e7f6
feat: changes #1
```

The changelog object ended up as 

```{
  changelog: '* feat: changes (59a134e)\n' +
    '* feat: changes #1 (74415e7)\n' +
    '* feat: changes #1 (b5d89f4)'
}
```

Then, from here https://github.com/release-it/release-it/blob/main/lib/plugin/github/GitHub.js#L405 and when it tries to export the issues using the [getResolvedIssuesFromChangelog](https://github.com/release-it/release-it/blob/main/lib/plugin/github/GitHub.js#L417) it cannot picked up any issues, and the reasons are:

1. If the only commit would be the first one, the one with a extra white line, the `#1` does not ended up on the changelog object to extract the issue but it does ends up on the CHANGELOG file, for example, https://github.com/juancarlosjr97/relese-automation-test/commit/8ef0da5b78af963072a752686c40e3d537371a10 is as follows
  
  ```
commit: 8ef0da5b78af963072a752686c40e3d537371a10
feat: add release configuration

Fixes #1
  ```

2. Missing action to commit for [issueParser](https://github.com/release-it/release-it/blob/main/lib/plugin/github/util.js#L34)

I modified locally the execution to see the changelog, so the first images shows a regular execution

<img width="1253" alt="Screenshot 2024-12-08 at 9 10 10 am" src="https://github.com/user-attachments/assets/fb15fdcc-a961-4728-9c65-3b34901d8c26">

The second image shows the changelog updated during execution by adding the call to action as follows on line 425

```
changelog = changelog.replace("(59a134e)", "fixes #1 (59a134e)");
```

As the commit `59a134e` has an extra white line.

<img width="1261" alt="Screenshot 2024-12-08 at 9 10 33 am" src="https://github.com/user-attachments/assets/a4505309-ac1d-4dcf-beca-d4cffe4d53e3">

Additionally, I also tried doing with multiline with the call to action to resolve by the parser, but given it does not finish on the same line given the changelog object it is still not picked up

```
feat: changes
  
Fixes #1
```

<img width="1259" alt="Screenshot 2024-12-08 at 9 15 20 am" src="https://github.com/user-attachments/assets/95ae99bb-f6d8-4199-bc1e-4bd47f0c5c31">

The execution of the four commits can be found here: https://github.com/juancarlosjr97/relese-automation-test/actions/runs/12220578048/job/34088558626

**Comment by juancarlosjr97** [comment]
So, I have not checked how to address the first issue, the reference to the issue is not found on the commit when it is multiline, but I think just taking the whole commit should do it.

I have focused on the second issue, which is the GitHub parser to get the issue to be resolved from the commit using the library https://www.npmjs.com/package/issue-parser.

**Comment by juancarlosjr97** [comment]
I am sure the first one can be resolved, the second one is slightly more tricky as we need to get the issue.

One idea is to get all those strings starting with `#` and try to do a comment on the issues found.

**Comment by juancarlosjr97** [comment]
Ok, so if I write a commit such as this one...

```
feat: adding changes for release
This is the body
Fixed #1
```

It is picked up successfully but as soon as white line is added, it is lost the reference to the issue after the first line.

**Comment by juancarlosjr97** [comment]
I think there is no issue with release-it, just how things work, so perhaps only a documentation update will be enough.

**Comment by webpro** [comment]
So if I understand correctly, it's likely an issue with `issue-parser`, right?

It could maybe be an option to remove empty lines before we hand it over to `issueParsers`. Might result in unexpected behavior as well, although vanishing commits is never what you want I guess.

**Comment by juancarlosjr97** [comment]
Yes, if we remove the empty lines before passing it to the `issueParser()`, it should be able to pick it up and add the comment to the issue.

We can handle this in two separate flows: one that does not affect the changelog process by handling the commit as usual, and a second flow dedicated only to identifying the issue and making the comment. This approach should prevent any interference or unexpected results.

**Comment by webpro** [comment]
Sounds like a plan! Is that something you would like to submit a PR for?

**Comment by juancarlosjr97** [comment]
Absolutely! I will see if I can work on it this weekend :)

**Comment by webpro** [comment]
This was in the core GitHub comments path rather than conventional-changelog or issue-parser.

The old implementation generated another changelog using the default `%s`-only Git log. That discarded commit bodies and footers before parsing, including a multiline `Fixes #1`.

Since release-it 19.2.3, the GitHub step reuses the already-generated changelog instead: https://github.com/release-it/release-it/commit/316dbfa458d670fc92d2da7fe7298ad90f44dc68

A retained `Fixes #1` footer is parsed correctly, so I'm closing this as fixed. Let me know if I'm wrong, happy to re-open.
